odoo/modules/loading.py

```diff
@@ -31,56 +31,36 @@
     from odoo.tests.result import OdooTestResult
     from .module_graph import ModuleNode
 
+    LoadMode = typing.Literal['init', 'update']
+    LoadKind = typing.Literal['data', 'demo']
+
 _logger = logging.getLogger(__name__)
 
 
-def load_data(env: Environment, idref, mode: str, kind: str, package: ModuleNode) -> bool:
+def load_data(env: Environment, idref, mode: LoadMode, kind: LoadKind, package: ModuleNode) -> bool:
     """
-
-    kind: data, demo, test, init_xml, update_xml, demo_xml.
-
-    noupdate is False, unless it is demo data or it is csv data in
-    init mode.
+    noupdate is False, unless it is demo data
 
     :returns: Whether a file was loaded
-    :rtype: bool
     """
+    keys = ('init_xml', 'data') if kind == 'data' else ('demo',)
+
+    files: set[str] = set()
+    for k in keys:
+        if k == 'init_xml' and package.manifest[k]:
+            _logger.warning("module %s: key 'init_xml' is deprecated in Odoo 19.", package.name)
```

Comment:
Maybe we could simply remove the support for `'init_xml'` and raise an error if `init_xml` is detected in `package.manifest`
It will simply the code a lot. And it is quite hard and boring to promise the behavior of `init_xml` is exactly the same after the refectoring.